src/js/server/schemaMiddleware.js

```diff
@@ -0,0 +1,172 @@
+        handleRequest: {
+            func: "{that}.sendResponse",
+            args: [400, "{that}.options.body"]
+        }
+    }
+});
+
+fluid.registerNamespace("gpii.schema.middleware");
+
+gpii.schema.middleware.rejectOrForward  = function (that, req, res, next) {
+    var toValidate = fluid.model.transformWithRules(req, that.options.rules.requestContentToValidate);
+    var results = that.validator.validate(that.options.schemaKey, toValidate);
+    if (results) {
+        var transformedResults = fluid.model.transformWithRules(results, that.options.rules.validationErrorsToResponse);
+        that.events.onInvalidRequest.fire(req, res, transformedResults);
+    }
+    else {
+        next();
+    }
+};
+
+/*
+
+    The `gpii.express.middleware` that fields invalid responses itself and passes valid ones through to the `next`
+    Express router or middleware function.
+
+ */
+fluid.defaults("gpii.schema.middleware", {
+    gradeNames: ["gpii.express.middleware", "gpii.express.requestAware.base", "gpii.hasRequiredOptions"],
+    requiredFields: {
+        "schemaPath": true,
+        "schemaKey": true
+    },
+    responseSchemaKey: "message.json",
+    responseSchemaUrl: "http://terms.raisingthefloor.org/schema/message.json",
+    messages: {
+        error: "The JSON you have provided is not valid."
+    },
+    rules: {
+        validationErrorsToResponse: {
+            "ok": {
+                literalValue: false
+            },
+            "message": {
+                literalValue: "{that}.options.messages.error"
+            },
+            "fieldErrors": ""
+        }
+    },
+    events: {
+        "onInvalidRequest": null
+    },
+    handlerGrades: ["gpii.schema.middleware.handler"],
+    dynamicComponents: {
+        requestHandler: {
+            createOnEvent: "onInvalidRequest",
+            options: {
+                body: "{arguments}.2",
+                schemaKey: "{gpii.schema.middleware}.options.responseSchemaKey",
+                schemaUrl: "{gpii.schema.middleware}.options.responseSchemaUrl"
+            }
+        }
+    },
+    components: {
+        validator: {
+            type: "gpii.schema.validator.ajv.server",
+            options: {
+                schemaPath: "{gpii.schema.middleware}.options.schemaPath"
+            }
+        }
+    },
+    invokers: {
+        middleware: {
+            funcName: "gpii.schema.middleware.rejectOrForward",
+            args:     ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+        }
+    }
+});
+
+/*
+    A wrapper for the `gpii.express.requestAware.router` grade that seamlessly wires in JSON Schema validation.
+ */
+fluid.defaults("gpii.schema.middleware.requestAware.router", {
+    gradeNames: ["gpii.express.router.passthrough"],
+    method:     "post",
+    rules: {
+        requestContentToValidate: {
+            "": "body"
+        }
+    },
+    components: {
+        // required middleware that provides `req.body`
+        json: {
```

Comment:
This whole grade seems too heavyweight as well as duplicating definitions which belong upstairs. We should have a call to discuss how it can be better factored and become more usable with less dependence on gpii-express router infrastructure.